app/javascript/components/enrollment/steps/Contact.js

```diff
@@ -188,10 +245,21 @@ class Contact extends React.Component {
         secondary_telephone: yup.string().phone().max(200, 'Max length exceeded, please limit to 200 characters.'),
         primary_telephone_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
         secondary_telephone_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
-        international_telephone: yup.string().max(50, 'Max length exceeded, please limit to 50 characters.').nullable(),
+        international_telephone: yup.string().max(50, 'Max length exceeded, please limit to 50 characters.'),
         email: yup.string().email('Please enter a valid Email.').max(200, 'Max length exceeded, please limit to 200 characters.'),
         confirm_email: yup.string().oneOf([yup.ref('email'), null], 'Confirm Email must match.'),
         preferred_contact_method: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_contact_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_contact_name: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_primary_telephone: yup.string().phone().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_secondary_telephone: yup.string().phone().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_primary_telephone_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_secondary_telephone_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_international_telephone: yup.string().max(50, 'Max length exceeded, please limit to 50 characters.'),
+        alternate_email: yup.string().email('Please enter a valid Email.').max(200, 'Max length exceeded, please limit to 200 characters.'),
+        confirm_alternate_email: yup.string().oneOf([yup.ref('alternate_email'), null], 'Confirm Email must match.'),
+        alternate_preferred_contact_method: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
+        alternate_preferred_contact_time: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
```

Comment:
since this method is getting kind of big and these validations are all the same, can we define these in a constant somewhere like:

```
alternate_contact_validations = {
        alternate_contact_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_contact_name: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_primary_telephone: yup.string().phone().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_secondary_telephone: yup.string().phone().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_primary_telephone_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_secondary_telephone_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_international_telephone: yup.string().max(50, 'Max length exceeded, please limit to 50 characters.'),
        alternate_email: yup.string().email('Please enter a valid Email.').max(200, 'Max length exceeded, please limit to 200 characters.'),
        confirm_alternate_email: yup.string().oneOf([yup.ref('alternate_email'), null], 'Confirm Email must match.'),
        alternate_preferred_contact_method: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        alternate_preferred_contact_time: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
}
```

and then here we can do:
```
schema = yup.object().shape({
        ...alternate_contact_validations,
        contact_type: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        contact_name: yup.string().max(200, 'Max length exceeded, please limit to 200 characters.'),
        primary_telephone: yup
```

in fact, maybe we should just put all the validated fields that don't change into this constant and call it `static_contact_validations` or something

Reply:
this was a great idea and cleans up the code a lot: https://github.com/SaraAlert/SaraAlert/pull/1116/commits/ce2385245655c4f5b21b0975725d722dbf0713df